pkg/kubernetes/controllerutil.go

```diff
@@ -163,3 +166,54 @@ func LookupResource(name, namespace string, resource runtime.Object, client clie
 	}
 	return nil, nil
 }
+
+// getOperatorNamespace returns the operator namespace or the WATCH_NAMESPACE
+// value if the operator is running outside the cluster.
+// "OSDK_FORCE_RUN_MODE"="local" must be set if running locally (not in the cluster)
+func GetOperatorNamespace() (string, error) {
+	operatorNs, err := k8sutil.GetOperatorNamespace()
+	// This makes everything works even running outside the cluster
+	if err == k8sutil.ErrRunLocal {
+		var operatorWatchNs string
+		operatorWatchNs, err = k8sutil.GetWatchNamespace()
+		if operatorWatchNs != "" {
+			operatorNs = strings.Split(operatorWatchNs, ",")[0]
+		}
+	}
+	return operatorNs, err
+}
+
+func LookupOperatorTokenSecret(client client.Client, logger logr.Logger) (*corev1.Secret, error) {
+	operatorName, err := k8sutil.GetOperatorName()
+	if err != nil {
+		return nil, err
+	}
+
+	operatorNamespace, err := GetOperatorNamespace()
+	if err != nil {
+		return nil, err
+	}
+
+	ctx := context.TODO()
+	serviceAccount := &corev1.ServiceAccount{}
+	if err := client.Get(ctx, types.NamespacedName{Namespace: operatorNamespace, Name: operatorName}, serviceAccount); err != nil {
+		return nil, fmt.Errorf("Unable to retrieve operator ServiceAccount: %w", err)
+	}
+
+	var secretRef *corev1.ObjectReference
+	for _, s := range serviceAccount.Secrets {
+		if strings.Contains(s.Name, "-token-") {
```

Comment:
@ryanemerson probably searching for all tokens in service account and validating for the `kubernetes.io/service-account.name` annotation give us a bit more guarantied result rather than name template.

Reply:
@ryanemerson , please have a look at the `oc sa get-token ` implementation
https://github.com/openshift/oc/blob/master/pkg/cli/serviceaccounts/gettoken.go
I also need this function for #1020

Reply:
`LookupServiceAccountTokenSecret` implemented inside #1020
@ryanemerson please have a look